explorer_backend/services/tutorials_db.ts

```diff
@@ -0,0 +1,60 @@
+import Database from "better-sqlite3";
+const db = new Database("./database.db");
+import fs from "node:fs/promises";
+import { v7 as uuidv7 } from 'uuid';
+
+export interface Tutorial {
+  id: string;
+  text: string;
+  contracts: string;
+  stage: number;
+}
+
+db.exec(
+  `
+    CREATE TABLE TUTORIALS (
+      id TEXT,
+      text TEXT,
+      contracts TEXT,
+      stage INTEGER
+    )
+  `,
+);
+
+const getStmt = db.prepare<[string, string, number], Tutorial>(
+  "SELECT id, text, contracts FROM TUTORIALS WHERE text = ? AND contracts = ? AND stage = ?",
+);
+
+const getByStageStmt = db.prepare<number, Tutorial>(
+  "SELECT id, text, contracts FROM TUTORIALS WHERE stage = ?",
+);
+
+export const getTutorial = (stage: number): Tutorial | null => {
+  return getByStageStmt.get(stage) || null;
+};
+
+export const setTutorial = (text: string, contracts: string, stage: number): Tutorial => {
+  const existing = getStmt.get(text, contracts, stage);
+  if (existing) {
+    return existing;
+  }
+
+  const updateStmt = db.prepare("UPDATE TUTORIALS SET stage = 0 WHERE stage = ?");
+  updateStmt.run(stage);
+
+  const id = uuidv7();
```

Comment:
It's better not generate this id here. Because it would be unstable and you couldn't update proper tutorial. Like If you want to modify or add step or change text.

Reply:
And insert into on conflict(id) replace could be easier in that case